components/movementsensor/gpsutils/vrs.go

```diff
@@ -100,8 +100,21 @@ func ConnectToVirtualBase(ctx context.Context, ntripInfo *NtripInfo,
 	// We currently only write the GGA message when we try to reconnect to VRS. Some documentation for VRS states that we
 	// should try to send a GGA message every 5-60 seconds, but more testing is needed to determine if that is required.
 	// get the GGA message from cached data
-	ggaMessage, err := getGGA()
-	if err != nil {
+	var ggaMessage string
+	success := false
+	attempts := 0
```

Comment:
I just finished getting rid of this antipattern in ntrip.go, over in https://github.com/viamrobotics/rdk/pull/4215! I think it's an antipattern because the for loop reads like a bunch of goto statements instead of either a traditional while loop or a traditional for loop, making the control flow harder to follow. 

If you split this out into a separate function (`getGGAWithRetries()`, though I wish I had a better name), then when you succeed, you can just return the GGA message from inside the for loop. Then, if you get past the for loop, you must have failed, and can return errors. This lets you get rid of `success` entirely, and then you can move all uses of `attempts` into the line defining the for loop itself, which is easier to understand. It also encapsulates this subtask (getting a GGA message even if it fails at first) into a separate function, which makes the code easier to chunk into smaller, more easily understood pieces.

```go
func getGGAWithRetries() (string, error) {
    for attempts := 0; attempts < maxConnectAttempts; attempts++ {
        ggaMessage, err = getGGA()
        if err == nil {
            return ggaMessage, err
        }
    }
    return "", err // Return the most recent error, maybe?
}
...
func connectToVirtualBase() {
    ...
    ggaMessage, err := getGGAWithRetries()
    // The rest of this function is unmodified
```

Reply:
(I realize I was not asked to be a reviewer on this PR, and I might be getting in the way. Leave as-is if you want. It'll either get cleaned up eventually, or it won't be, and frankly any of those outcomes is fine.)